apps/random/rng.c

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2022, 2023 - Tillitis AB
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include <lib.h>
+#include <tk1_mem.h>
+#include <types.h>
+
+#include "rng.h"
+
+// clang-format off
+static volatile	uint32_t *cdi =          (volatile uint32_t *)TK1_MMIO_TK1_CDI_FIRST;
+static volatile uint32_t *trng_status  = (volatile uint32_t *)TK1_MMIO_TRNG_STATUS;
+static volatile uint32_t *trng_entropy = (volatile uint32_t *)TK1_MMIO_TRNG_ENTROPY;
+
+#define RESEED_TIME 1000
+// clang-format on
+
+uint8_t rng_initalized = 0;
+
+static uint32_t entropy_get()
+{
+	while ((*trng_status & (1 << TK1_MMIO_TRNG_STATUS_READY_BIT)) == 0) {
+	}
+	return *trng_entropy;
+}
+
+// Do you really keep the entropy for byte 32-63?
```

Comment:
What does this mean?

Reply:
This was a memory note to discuss with Joachim. It is taken care of so I'm removing it in the new branch.